### DISCUSSION TITLE
OTA support

### DISCUSSION THREAD

**Discussion by biemster** [holysnippet/pico_eth#6] [Ideas]
I'm going to build a project which could definitely use OTA (or should it be called OTW?) so I can upload new firmware using a web page hosted on the Pico. I believe arduino-pico for the pico W has a similar feature integrated in their IDE.
Would this be useful to integrate here? (so would you like a PR when it is done?)

**Comment by holysnippet** [comment]
It's a good idea, I thought about it but I kind of abandoned the idea. I already did it on PIC24 and to save the FLASH space I should have only one copy of the stack. On PIC it meant to modify the linker script and the sources. I don't know how to do it on Pico and if I did it (personally) I would do it like this.

I'm not against this idea at all and I might even have a use for such a program but I don't know if it belongs with the Ethernet driver. This program would be a dependency of lwIP.

But on the other hand it would open up the possibilities and make it easier to access and use.

It should be written in such a way that it is only included if it is needed.

**Reply by biemster** [reply]
> It should be written in such a way that it is only included if it is needed.

Yeah fully agree with that.
I was reading up a bit on it, and was thinking of using
https://github.com/rhulme/pico-flashloader
as a starting point, maybe using
https://github.com/usedbytes/picowota
as a guide.
Also https://forums.raspberrypi.com/viewtopic.php?t=335757 has a lot of nice ideas.
